ContainerHandling/traefik/Create-CustomTraefikImage.ps1

```diff
@@ -0,0 +1,135 @@
+ .Description
+  Create a custom Traefik docker image, based on the best generic image version according to your Windows version.
+  This is useful when using Windows Server 2022 for example, where no prebuilt image is provided.
+ .Parameter imageName
+  Name of the docker image, which will be created. Default value is "mytraefik"
+ .Parameter doNotUpdateConfig
+  Specifies to only create the traefik image, but not updating the BcContainerHelper configuration to use this Traefik image.
+ .Example
+  Create-CustomTraefikImage -imageName mytraefik
+#>
+function Create-CustomTraefikImage {
+    [CmdletBinding()]
+    Param (
+        [string] $imageName = "mytraefik",
+        [switch] $doNotUpdateConfig
+    )    
+
+    Process {
+      $originalPath = Get-Location
+
+      function PullDockerImage {
+          Param(
+              [Parameter(Mandatory=$true)]
+              [string]$imageName
+          )
+
+          $result = $true
+          $arguments = ("pull $imageName")
+          $pinfo = New-Object System.Diagnostics.ProcessStartInfo
+          $pinfo.FileName = "docker.exe"
+          $pinfo.RedirectStandardError = $true
+          $pinfo.RedirectStandardOutput = $true
+          $pinfo.CreateNoWindow = $true
+          $pinfo.UseShellExecute = $false
+          $pinfo.Arguments = $arguments
+          $p = New-Object System.Diagnostics.Process
+          $p.StartInfo = $pinfo
+          $p.Start() | Out-Null
+
+          $outtask = $null
+          $errtask = $p.StandardError.ReadToEndAsync()
+          $out = ""
+          $err = ""
+
+          do {
+              if ($null -eq $outtask) {
+                  $outtask = $p.StandardOutput.ReadLineAsync()
+              }
+              $outtask.Wait(100) | Out-Null
+              if ($outtask.IsCompleted) {
+                  $outStr = $outtask.Result
+                  if ($null -eq $outStr) {
+                      break
+                  }
+                  if (!$silent) {
+                      Write-Host $outStr
+                  }
+                  $out += $outStr
+                  $outtask = $null
+              } elseif ($outtask.IsCanceled) {
+                  break
+              } elseif ($outtask.IsFaulted) {
+                  break
+              }
+          } while(!($p.HasExited))
+
+          $err = $errtask.Result
+          $p.WaitForExit();
+
+          if ($p.ExitCode -ne 0) {
+              $result = $false
+              if (!$silent) {
+                  $err = $err.Trim()
+                  if ("$error" -ne "") {
+                      Write-Host $error -ForegroundColor Red
+                  }
+                  Write-Host "ExitCode: "+$p.ExitCode -ForegroundColor Red
+                  Write-Host "Commandline: docker $arguments" -ForegroundColor Red
+              }
+          }
+          return $result
+      }
+
+
+      try {
+          $bestGenericImage = Get-BestGenericImageName
+          $servercoreVersion = $bestGenericImage.Split(':')[1]
```

Comment:
BestGenericImageName might be 

`mcr.microsoft.com/businesscentral:10.0.20348.1668-dev
`

So you would need $bestGenericImage.Split(':')[1].Split('-')[0]